src/sample.jl

```diff
@@ -65,12 +65,13 @@ function mcmcsample(
     progressname = "Sampling",
     callback = nothing,
     discard_initial = 0,
+    thinning = 1,
     chain_type::Type=Any,
     kwargs...
 )
     # Check the number of requested samples.
     N > 0 || error("the number of samples must be ≥ 1")
-    Ntotal = N + discard_initial
+    Ntotal = thinning * N + discard_initial
```

Comment:
I think this is not correct and causes the incorrect progress bars:
```suggestion
    Ntotal = thinning * (N - 1) + discard_initial + 1
```

Reply:
Ah. That makes sense because the last few iterations aren't saved. Thanks!